x-pack/plugins/security_solution/server/lib/detection_engine/rule_monitoring/logic/detection_engine_health/event_log/aggregations/rule_execution_stats.ts

```diff
@@ -26,23 +26,33 @@ import {
 import { DEFAULT_PERCENTILES } from '../../../utils/es_aggregations';
 import type { RawData } from '../../../utils/normalization';
 import * as f from '../../../event_log/event_log_fields';
+import {
+  ALERTING_PROVIDER,
+  RULE_EXECUTION_LOG_PROVIDER,
+} from '../../../event_log/event_log_constants';
 
 export type RuleExecutionStatsAggregationLevel = 'whole-interval' | 'histogram';
 
 export const getRuleExecutionStatsAggregation = (
-  aggregationContext: RuleExecutionStatsAggregationLevel
+  aggregationLevel: RuleExecutionStatsAggregationLevel
 ): Record<string, estypes.AggregationsAggregationContainer> => {
   return {
-    totalExecutions: {
-      cardinality: {
-        field: f.RULE_EXECUTION_UUID,
-      },
-    },
     executeEvents: {
       filter: {
-        term: { [f.EVENT_ACTION]: 'execute' },
+        bool: {
+          filter: [
+            { term: { [f.EVENT_PROVIDER]: ALERTING_PROVIDER } },
+            { term: { [f.EVENT_ACTION]: 'execute' } },
+            { term: { [f.EVENT_CATEGORY]: 'siem' } },
+          ],
+        },
       },
       aggs: {
+        totalExecutions: {
+          cardinality: {
+            field: f.RULE_EXECUTION_UUID,
+          },
+        },
```

Comment:
Moving this cardinality aggregation under `executeEvents` is also important to correctly calculate the number of executions of _detection_ rules.